assets/js/components/OptIn.js

```diff
@@ -39,7 +39,16 @@ import Link from './Link';
 import useViewContext from '../hooks/useViewContext';
 const { useSelect, useDispatch } = Data;
 
-export default function OptIn( { id, name, className } ) {
+export default function OptIn( props ) {
+	const viewContext = useViewContext();
+
+	const {
+		id = 'googlesitekit-opt-in',
+		name = 'optIn',
+		className,
+		trackEventCategory = viewContext,
+	} = props;
```

Comment:
Moving the default props into the function and out of "`.defaultProps` is nice, but I think they should go in the function definition rather than here. Technically accessing them via `props.id` elsewhere in the function would provide no default and I think be a bit of an unexpected behaviour. I know we don't tend to do that, but also since this isn't the first line in the function body it makes the default args a bit less obvious. Also, I think things like VSCode would be less likely to infer the default args here.

Let's go with something like:

```suggestion
export default function OptIn( {
	id = 'googlesitekit-opt-in',
	name = 'optIn',
	className,
	trackEventCategory = viewContext,
} ) {
	const viewContext = useViewContext();
```

instead.

Reply:
Updated 👍